src/nvim/ui.c

```diff
@@ -441,8 +441,12 @@ void ui_puts(uint8_t *str)
 
     if (p_wd) {  // 'writedelay': flush & delay each time.
       ui_flush();
-      uint64_t wd = (uint64_t)labs(p_wd);
-      os_delay(wd, false);
+      if (!((got_int || (did_emsg && force_abort) || current_exception))){
```

Comment:
Not sure about the other conditions besides `got_int` here: wouldn't that skip the delay in case of `echoerr` etc, which shouldn't be the case?